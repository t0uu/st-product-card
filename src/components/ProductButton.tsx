import React, { useContext, CSSProperties,useCallback } from 'react';
import { ProductContext } from './ProductCard';
import styles from '../styles/styles.module.css'

interface Props{
    className?: string;
    style?: CSSProperties;
    
}

export const ProductButtons = ({className,style}: Props) => {
    // TODO: maxCount
    const {counter,increaseBy,maxCount} = useContext(ProductContext);

    // TODO: isMaxReached = useCallback, dependecias [count,maxCount] // TRUE O FALSE
    // true si el count === maxCount
    // false si no lo es
    const count = useCallback(
      () => {
        
        if(counter === maxCount){
            return true
        }else{
            return false
        }
      },
      [counter,maxCount]
    )
     
    return (
<div className={`${styles.buttonsContainer} ${className}`} style={style}>
            <button onClick={() => increaseBy(-1)} className={styles.buttonMinus}>-</button>

            <div className={styles.countLabel}>{counter}</div>
            <button onClick={() => increaseBy(1)} className={  `${styles.buttonAdd}  ${ count() && styles.disabled}`  }>+</button>
        </div>
    )
}