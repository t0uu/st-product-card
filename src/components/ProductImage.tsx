import React,{ useContext, CSSProperties } from 'react';
import styles from '../styles/styles.module.css'
import noImage from '../assets/no-image.jpg'
import { ProductContext } from './ProductCard';


export interface Props {
    img?: string;
    className?: string;
    style?: CSSProperties
}

export const ProductImage = ({img = '',className,style} : Props) => {

    const {product} = useContext(ProductContext)
    let imgToShow: string;

    if(img){
        imgToShow = img;
    }else if (product.img){
        imgToShow = product.img
    }else {
        imgToShow = noImage;
    }
    return (
        <img style={style} className={`${styles.productImg} ${className}`} src={imgToShow} alt="Product image"/>
    )
}