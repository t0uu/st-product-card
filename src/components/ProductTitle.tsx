import { ProductContext } from './ProductCard';
import React,{ useContext, CSSProperties } from 'react';
import styles from '../styles/styles.module.css'


export interface Props {
    title?:string,
    className?:string
    style?: CSSProperties
}

export const ProductTitle = ({title,className,style} : Props) => {

    const { product} = useContext(ProductContext)


    return (
        <span  style={style} className={`${styles.productDescription} ${className}`}>{ title ? title : product.title}</span>
    )
}