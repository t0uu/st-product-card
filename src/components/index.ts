import { ProductCard as ProductCardHOC } from './ProductCard';
import { ProductTitle } from './ProductTitle';
import { ProductImage } from './ProductImage';
import { ProductButtons } from './ProductButton';
import { ProductCardHOCProps } from '../interfaces/interfaces';
export {ProductImage} from './ProductImage'
export {ProductTitle} from './ProductTitle'
export {ProductButtons} from './ProductButton'


export const ProductCard:ProductCardHOCProps = Object.assign(ProductCardHOC,{
    Title: ProductTitle,
    Image: ProductImage,
    Buttons: ProductButtons
})




export default ProductCard;