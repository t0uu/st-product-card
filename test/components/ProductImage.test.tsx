import React from 'react'
import renderer from 'react-test-renderer'
import { ProductImage,ProductCard } from '../../src/components';
import {  product2 } from './data/product';


describe('ProductTitle', () => {

    test('debe de mostrar el componente correctamente con la imagen', () => {
        const wrapper = renderer.create(
            <ProductImage  img={product2.img}/>
        )
        expect(wrapper.toJSON()).toMatchSnapshot();

    });

    test('debe de mostrar el product card con la imagen' , () => {
        const wrapper = renderer.create(
            <ProductCard
            product={product2}
            >
                {
                    () => (
                       <ProductImage img={product2.img}/>
                    )
                }

            </ProductCard>
        )
        expect(wrapper.toJSON()).toMatchSnapshot();
    })
})