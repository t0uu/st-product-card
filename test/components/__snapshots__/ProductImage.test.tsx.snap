// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe de mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductTitle debe de mostrar el componente correctamente con el titulo personalizado 1`] = `
<img
  alt="Product image"
  className="productImg undefined"
  src="./coffee-mug.png"
/>
`;

exports[`ProductTitle debe de mostrar el componente correctamente con la imagen 1`] = `
<img
  alt="Product image"
  className="productImg undefined"
  src="./coffee-mug.png"
/>
`;

exports[`ProductTitle debe de mostrar el product card con la imagen 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
